### PR TITLE
Add validate integration/script file path

### PR DESCRIPTION
**Pull request by ChanochShayner**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
related: https://github.com/demisto/etc/issues/29240

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by guyfreund — COMMENTED** [review]
@ChanochShayner Is this also refactors the content repo to be aligned with this fix? is this also enhances the format command?

File: demisto_sdk/commands/common/errors.py [hunk 1]
```diff
@@ -67,9 +67,11 @@
     "multiple_default_arg": "IN134",
     "missing_output_context": "IN136",
     "invalid_integration_parameters_display_name": "IN135",
+    "is_valid_integration_file_path": "IN136",
```
Comment (roysagi) [inline demisto_sdk/commands/common/errors.py:70]: I think @JasBeilin is using IN136 if you merge from master you should see that.

File: demisto_sdk/commands/common/errors.py [hunk 2]
```diff
@@ -467,6 +469,12 @@ def invalid_integration_parameters_display_name(invalid_display_names):
                "Integration parameters display name should be capitalized and spaced using whitespaces " \
                "and not underscores ( _ )."
 
+    @staticmethod
+    @error_code_decorator
+    def is_valid_integration_file_path(integration_file):
```
Comment (roysagi) [inline demisto_sdk/commands/common/errors.py:474]: This error is good for packaged integrations but we should create a different one for unified integrations (`integration-something.yml`) cause we still have those.

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 3]
```diff
@@ -935,6 +937,29 @@ def is_valid_parameters_display_name(self) -> bool:
...
 
+    def is_valid_integration_file_path(self) -> bool:
+        absolute_file_path = self.file_path
+        integrations_folder = os.path.dirname(absolute_file_path).split('/')[-1]
+        integration_file = os.path.basename(absolute_file_path)
+
+        integration_file, _ = os.path.splitext(integration_file)
+
+        if integrations_folder == 'Integrations':
+            if integration_file.startswith('integration-'):
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/integration.py:948]: I think your logic is off here,  you want to check if this is not occurring and return False in that case.

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 4]
```diff
@@ -935,6 +937,29 @@ def is_valid_parameters_display_name(self) -> bool:
 
         return True
 
+    def is_valid_integration_file_path(self) -> bool:
+        absolute_file_path = self.file_path
+        integrations_folder = os.path.dirname(absolute_file_path).split('/')[-1]
+        integration_file = os.path.basename(absolute_file_path)
+
+        integration_file, _ = os.path.splitext(integration_file)
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/integration.py:945]: add a comment that you are dropping the file extension.

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 5]
```diff
@@ -935,6 +937,29 @@ def is_valid_parameters_display_name(self) -> bool:
 
         return True
 
+    def is_valid_integration_file_path(self) -> bool:
+        absolute_file_path = self.file_path
+        integrations_folder = os.path.dirname(absolute_file_path).split('/')[-1]
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/integration.py:942]: I think there should be a way to use `os` only to get the dir name without the split - the reason being that this does not support windows in this way.

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 6]
```diff
@@ -935,6 +937,29 @@ def is_valid_parameters_display_name(self) -> bool:
...
+        integration_file = os.path.basename(absolute_file_path)
+
+        integration_file, _ = os.path.splitext(integration_file)
+
+        if integrations_folder == 'Integrations':
+            if integration_file.startswith('integration-'):
+                return True
+
+        if integration_file == integrations_folder:
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/integration.py:951]: same thing - I think your logic is off.

File: demisto_sdk/commands/common/hook_validations/script.py [hunk 7]
```diff
@@ -220,3 +223,25 @@ def is_valid_as_deprecated(self) -> bool:
...
+        scripts_folder = os.path.dirname(absolute_file_path).split('/')[-1]
+        script_file = os.path.basename(absolute_file_path)
+        script_file, _ = os.path.splitext(script_file)
+
+        if scripts_folder == 'Scripts':
+            if script_file.startswith('script-'):
+                return True
+
+        if script_file == scripts_folder:
+            return True
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/script.py:238]: same comments as in integration

**Review by roysagi — CHANGES_REQUESTED** [review]

File: demisto_sdk/commands/common/errors.py (hunk 1 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/errors.py:70]: done

File: demisto_sdk/commands/common/errors.py (hunk 2 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/errors.py]: done

File: demisto_sdk/commands/common/hook_validations/integration.py (hunk 3 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/integration.py]: done

File: demisto_sdk/commands/common/hook_validations/integration.py (hunk 4 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/integration.py:943]: done

File: demisto_sdk/commands/common/hook_validations/integration.py (hunk 6 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/integration.py]: done

File: demisto_sdk/commands/common/hook_validations/script.py (hunk 7 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/script.py]: done

File: demisto_sdk/commands/common/hook_validations/integration.py (hunk 5 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/integration.py]: done

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 8]
```diff
@@ -934,6 +934,33 @@ def is_valid_parameters_display_name(self) -> bool:
...
+        integration_file, _ = os.path.splitext(integration_file)
+
+        if integrations_folder == 'Integrations':
+            if not integration_file.startswith('integration-'):
+
+                error_message, error_code = \
+                    Errors.is_valid_integration_file_path_in_integrations_folder(integration_file)
+                if self.handle_error(error_message, error_code, file_path=self.file_path):
+                    return False
+            return True
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/integration.py:952]: drop this return and change the if in 954 to elif

File: demisto_sdk/commands/common/hook_validations/script.py [hunk 9]
```diff
@@ -220,3 +223,27 @@ def is_valid_as_deprecated(self) -> bool:
...
+        script_file = os.path.basename(absolute_file_path)
+        script_file, _ = os.path.splitext(script_file)
+
+        if scripts_folder == 'Scripts':
+            if not script_file.startswith('script-'):
+
+                error_message, error_code = Errors.is_valid_script_file_path_in_scripts_folder(script_file)
+                if self.handle_error(error_message, error_code, file_path=self.file_path):
+                    return False
+            return True
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/script.py:239]: drop this return and change the if in 241 to elif

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,10 +1,11 @@
 # Changelog
 * Fixed an issue where downloading indicator layoutsconatiner objects failed.
 * Added a validation that makes sure outputs in integrations are matching the README file.
-* Fixed an issue when files were classified incorrectly when running **update-release-notes**.
 * Fixed an issue where the *create-id-set* flag in the **validate** command did not work.
 * Added a warning in case no id_set file is found when running the **validate** command.
 * Fixed an issue where changed files were not recognised correctly on forked branches in the **validate** and the **update-release-notes** commands.
+* Fixed an issue when files were classified incorrectly when running *update-release-notes*.
+* Added a validation that the integration and script file path's compatible with our convention.
```
Comment (roysagi) [inline CHANGELOG.md:8]:
```suggestion
* Added a validation that integration and script file paths are compatible with our convention.
```

**Review by roysagi — APPROVED** [review]
Please see my comments

**Comment by roysagi** [comment]
> @ChanochShayner Is this also refactors the content repo to be aligned with this fix? is this also enhances the format command?

I don't think this will fix the format command but it will fail validation and tell you to change your path so I'm not sure this is necessary.

File: demisto_sdk/commands/common/hook_validations/integration.py (hunk 8 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/integration.py:952]: done

File: demisto_sdk/commands/common/hook_validations/script.py (hunk 9 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/script.py:239]: done

File: CHANGELOG.md (hunk 10 above)
Comment (ChanochShayner) [reply CHANGELOG.md:8]: done

**Pull request merged**